R/data-combined.R

```diff
@@ -0,0 +1,680 @@
+      self$setDataTransformations(names)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    #' @description
+    #'
+    #' Transform raw data with required offsets and scale factors.
+    #'
+    #' @param xOffsets,yOffsets,xScaleFactors,yScaleFactors Either a single
+    #'   numeric value or a list of numeric quantities specifying offsets and
+    #'   scale factors to apply to raw values. The default offset is `0`, while
+    #'   default scale factor is `1`, i.e., the data will not be modified. If a
+    #'   list is specified, it should be the same length as `names` argument.
+    #'
+    #' @details
+    #'
+    #' A dataframe with respective raw quantities transformed using specified
+    #' offset and scale factor values.
+    #'
+    #' - For X and Y variables:
+    #'   `newValue = (rawValue + offset) * scaleFactor`
+    #'
+    #' - For error term:
+    #'   `newErrorValue = rawErrorValue * scaleFactor`
+
+    setDataTransformations = function(names = NULL,
+                                      xOffsets = 0,
+                                      yOffsets = 0,
+                                      xScaleFactors = 1,
+                                      yScaleFactors = 1) {
+      # check that the arguments to parameters make sense
+      xOffsets <- validateVectorArgs(xOffsets, type = "numeric")
+      yOffsets <- validateVectorArgs(yOffsets, type = "numeric")
+      xScaleFactors <- validateVectorArgs(xScaleFactors, type = "numeric")
+      yScaleFactors <- validateVectorArgs(yScaleFactors, type = "numeric")
+      names <- validateVectorArgs(names, type = "character")
+
+      # apply specified data transformations
+      private$.dataCombined <- private$.dataTransform(
+        data          = private$.dataCombined,
+        names         = names,
+        xOffsets      = xOffsets,
+        yOffsets      = yOffsets,
+        xScaleFactors = xScaleFactors,
+        yScaleFactors = yScaleFactors
+      )
+
+      # update private field with transformation values
+      private$.dataTransformations <- private$.extractTransforms(private$.dataCombined)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    ## getter methods ---------------
+
+    #' @description
+    #'
+    #' A method to extract a dataframe of simulated and/or observed data
+    #' (depending on instances of which classes have been added to the object).
+    #'
+    #' Note that the order in which you enter different object doesn't matter
+    #' because the returned dataframe is arranged alphabetically by dataset
+    #' name.
+    #'
+    #' @return
+    #'
+    #' In the returned dataframe, the following columns will always be present:
+    #'
+    #' name - group - dataType - xValues - xDimension - xUnit - yValues -
+    #' yErrorValues - yDimension - yUnit - yErrorType - yErrorUnit - molWeight
+    #'
+    #' @note
+    #'
+    #' The molecular weight (in `molWeight` column) is in `g/mol` units.
+
```

Comment:
Why do you leave an additional empty line (or even multiple empty lines) between the documentation and the function? This is not the default behavior of "generate roxygen code"?

I liked our "old" style better, this style makes the documentation look a bit "detached" from the function and also introduces additional lines of code.

Reply:
And it is not consistent... so definitely not something automated.